src/auth.rs

```diff
@@ -258,6 +258,31 @@ impl DeviceCodes {
     }
 }
 
+/// Exchange a code for a user access token
+///
+/// see: https://docs.github.com/en/developers/apps/identifying-and-authorizing-users-for-github-apps
+pub async fn get_access_token(
```

Comment:
Is there a reason this is a free function as opposed to a impl method?

Reply:
Hi @XAMPPRocky, TBH I didn't see any value in adding an additional structure as it was done for the device flow. In that case, it's useful as the params are received from another REST API. But if you see any case in which I'm not considering I can change this to be a method

Reply:
Ok, it seems that in octokit is defined as a struct:

export type ExchangeWebFlowCodeGitHubAppOptions = {
  clientType: "github-app";
  clientId: string;
  clientSecret: string;
  code: string;
  redirectUrl?: string;
  request?: RequestInterface;
};

https://github.com/octokit/oauth-methods.js/blob/main/src/exchange-web-flow-code.ts

Also the redirectUri seems to be opt.

I'll update the MR and replicate this behavior

Reply:
Sorry I meant how does it look to get the access token? Like what does the actual method call look like if you used octokit?

Reply:
There are two options (GitHub App is the recommended one):

```
export async function exchangeWebFlowCode(
  options: ExchangeWebFlowCodeOAuthAppOptions,
): Promise<ExchangeWebFlowCodeOAuthAppResponse>;

export async function exchangeWebFlowCode(
  options: ExchangeWebFlowCodeGitHubAppOptions,
): Promise<ExchangeWebFlowCodeGitHubAppResponse>;
```

https://github.com/octokit/oauth-methods.js/blob/main/src/exchange-web-flow-code.ts